### ISSUE TITLE
Standardization of AHK hotkeys/toggles

### ISSUE THREAD

**Issue by niboty** [llamositopia/kitboga-codeville#18]
So with the increasing amount of AHK scripts that are being developed for use on the stream having different hotkeys for all of them will become impossible to remember. I suggest we standardize the keys used in hotkeys as well as what hotkeys/functionality a script requires. 

The following are my initial suggestions, feel free to give input!

Standardized Hotkeys:
- 3 key input, 1st being standard, 2nd being script specific, 3rd being script options/toggles
- For the 1st key I suggest something on the lower left like Alt
- For the 2nd key it should probably be within 1 hand range of the 1st key, I suggest the left half of the letters on the keyboard.
- For the 3rd key it should have enough options for the script, I suggest any of the numbers. 

Functionality All Scripts should have:
- Standardized Exit app hotkey so Kitboga could shut down all scripts at once. ex: Ctrl + Esc
- Additional Script Specific exit app hotkey ex. Alt + [ScriptSpecificKey] + Esc
- Ability to Pause/stop ex. Alt + [ScriptSpecificKey] + 0
- Any options that the script has on functionality should be broken into the most common configurations/settings and be able to be toggled between using Alt + [ScriptSpecificKey] + [1-9]

Let me know what you think. With hotkeys I feel like we need to consider accidental triggers (3 keys should be good enough), memorability, and the ability to actually press all the keys

**Comment by llamositopia** [comment]
I definitely like this idea. It ensures scripts can work together without any conflicts. 👍

**Comment by yukihyou** [comment]
I think this is something that should be written RFC-like, although I think it would be best to get a bunch more scripts in here, then make the standards based on the use cases, alongside a PR to update them all at the same time. That way, a defacto standard might emerge, and save time on investigating options at this point.

```
Scripts MUST implement [whatever] as an immediate exit key. This key MUST NOT be blocked by the script to allow it to be handled by all running scripts.
Scripts MAY implement [x, y, z]
etc
```

**Comment by niboty** [comment]
Yeah good advice I'm down for some more official documentation, was just throwing out some ideas. We should get the rest of the official Kitboga scripts and finish up any of the existing AHK PRs. Then make sure a standard works for those and any other use cases we might have. 

And since Kit would be the one using it though we should probably get his input. Not sure how active he is with this and the best way to get in touch with him. @llamositopia any ideas?

But I'm thinking it will enable us to package together all the scripts and give it to Kit with some documentation. Then he can have all the scripts on one VM save and not worry about compatibility issues allowing him to quickly turn them on and off if he decides it might be a funny time for one, It will also be easier to keep the VM updated with the newest codeville works.

**Comment by adam6806** [comment]
I'm definitely interested in making a "master script" and standardizing all current scripts. During some initial research, I did find out that AHKPanic() is a function that can suspend all active scripts. I'm a java software engineer myself and I was pretty surprised to find out the number of things that can be done with AHK. Would be a fun project for sure and a fun way to contribute something to the stream. I'll start work and submit a PR when I have a working prototype. If someone else is actively developing this please let me know as I don't want to step on anyone's toes.

**Comment by adam6806** [comment]
There could also be a concept of a grouping of scripts. Kit could configure a group of scripts he wants to run and one hotkey could switch all running scripts off and start the specific scripts in that group. Might be helpful.
